sdk/src/util/driftClock/driftClock.ts

```diff
@@ -0,0 +1,26 @@
+import { ClockSubscriber } from "./baseClockSubscriber";
+
+/**
+ * Best-Effort "clock" for tracking the on chain slot and timestamp. 
+ * 
+ * Can subscribe in two ways: 
+ * - Self-contained subscriber that will poll the chain for the current slot and timestamp
+ * - Passive subscriber which other components can pipe their updates into
+ */
+export class DriftClock {
```

Comment:
it might make sense to track latest blockHeight too since that is the number you need to compare to check that a tx has expired (not slot)